pkg/controller/deployment/deployment_controller.go

```diff
@@ -507,6 +496,51 @@ func (dc *DeploymentController) syncDeploymentStatus(allRCs []*api.ReplicationCo
 	return nil
 }
 
+func (dc *DeploymentController) getNewRCAndOldRCs(deployment extensions.Deployment) (*api.ReplicationController, []*api.ReplicationController, error) {
+	oldRCs, err := dc.getOldRCs(deployment)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	maxOldV, err := maxRevision(oldRCs)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	newRC, err := dc.getNewRC(deployment, maxOldV)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if newRC.Annotations != nil && newRC.Annotations[deploymentutil.RevisionAnnotation] == "" &&
+		deployment.Annotations == nil || deployment.Annotations[deploymentutil.RevisionAnnotation] != newRC.Annotations[deploymentutil.RevisionAnnotation] {
```

Comment:
`||` has the lowest operator precedence, so the last relational expression will be evaluated in the case that any of the preceding ones fail, which isn't right. The expressions on this line need to be parenthesized.